pkg/certgenerator/v1beta1/generator.go

```diff
@@ -64,26 +68,54 @@ var _ manager.Runnable = &CertGenerator{}
 var _ manager.LeaderElectionRunnable = &CertGenerator{}
 
 func (c *CertGenerator) Start(ctx context.Context) error {
-	if err := c.generate(ctx); err != nil {
+	if c.enableGenerateCerts {
+		if err := c.generate(ctx); err != nil {
+			return err
+		}
+	}
+	// We need to wait for certs to get ready even though using cert-manager.
+	klog.Info("Waiting for certs to get ready.")
+	if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: time.Second,
+		Factor:   2,
+		Jitter:   1,
+		Steps:    10,
+		Cap:      time.Minute * 5,
+	}, ensureCertMounted); err != nil {
 		return err
 	}
 	// Sending an empty data to a certsReady means it starts to register controllers to the manager.
 	c.certsReady <- struct{}{}
 	return nil
 }
 
+// ensureCertMounted ensures that the generated certs are mounted inside the container.
+func ensureCertMounted(context.Context) (bool, error) {
+	certFile := filepath.Join(consts.CertDir, serverCertName)
+	if _, err := os.Stat(certFile); err != nil {
+		return false, nil
+	}
+	keyFile := filepath.Join(consts.CertDir, serverKeyName)
+	if _, err := os.Stat(keyFile); err != nil {
+		return false, nil
+	}
```

Comment:
Should we add the log line here as well ? E.g.
```
klog.Info("Public key file %v doesn't exist in the container yet", certFile)
klog.Info("Private key file %v doesn't exist in the container yet", keyFile)
```

Reply:
I hesitate to add the log since the logs will be output in large quantities and the controller's logs will be dirty.
So output logs every 30 sec or 15 sec might be better. 
@andreyvelich WDYT?

Reply:
Sure, that also works, if we can print them every 15 sec.